src/sentry/rules/history/preview.py

```diff
@@ -438,6 +442,7 @@ def apply_frequency_conditions(
             passes = [True] * len(activities)
             for conditions in condition_types.values():
                 # reuse frequency buckets for conditions of the same type
+                assert isinstance(conditions[0], BaseEventFrequencyCondition)
```

Comment:
does asserting on `conditions[0]` make mypy infer that the rest of the conditions are also `BaseEventFrequencyCondition`?

Reply:
hmm okay 🤔